indexer/indexer_tests/osm_editor_test.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "generator/generator_tests_support/test_feature.hpp"
+#include "generator/generator_tests_support/test_mwm_builder.hpp"
+
+#include "platform/local_country_file_utils.hpp"
+
+#include "storage/country_info_getter.hpp"
+
+#include "indexer/mwm_set.hpp"
+
+#include "map/feature_vec_model.hpp"
+
+namespace tests
+{
+class EditorTest
+{
+public:
+  EditorTest();
+  ~EditorTest();
+
+  void GetFeatureTypeInfoTest();
+  void GetEditedFeatureTest();
+  void GetEditedFeatureStreetTest();
+  void OriginalFeatureHasDefaultNameTest();
+  void GetFeatureStatusTest();
+  void IsFeatureUploadedTest();
+  void DeleteFeatureTest();
+  void ClearAllLocalEditsTest();
+  void GetFeaturesByStatusTest();
+  void OnMapDeregisteredTest();
+
+private:
+  template <typename TBuildFn>
+  MwmSet::MwmId ConstructTestMwm(TBuildFn && fn)
+  {
+    generator::tests_support::TestCity london(m2::PointD(1, 1), "London", "en", 100 /* rank */);
+    BuildMwm("TestWorld", feature::DataHeader::world, [&london](generator::tests_support::TestMwmBuilder & builder)
```

Comment:
The world should be built once. Same for SomeCountry. I suggest you to build world in the class ctor, and modify the signature of the method:
```c++
template <typename TBuilder>
MwmSet::MwmId ConstructTestCountry(string const & name, TBuilder && builder)
{
  return BuildMwm(name, country, forward<TBuilder>(builder));
}
```